src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala

```diff
@@ -215,12 +236,9 @@ object ConductRSandbox extends AutoPlugin {
 
     val logLevelEnv = Map("AKKA_LOGLEVEL" -> logLevelValue)
     val syslogEnv = cond0Ip.map(ip => Map("SYSLOG_IP" -> ip)).getOrElse(Map.empty)
-    val conductrFeaturesArgs =
-      if (featureNames.isEmpty)
-        Map.empty
-      else
-        Map("CONDUCTR_FEATURES" -> featureNames.mkString(","))
-    val envsArgs = (logLevelEnv ++ syslogEnv ++ envsValue ++ conductrFeaturesArgs).flatMap { case (k, v) => Seq("-e", s"$k=$v") }.toSeq
+    val conductrFeatureArgs = featureNames.toMapWithKey("CONDUCTR_FEATURES")(_.mkString(","))
```

Comment:
By all means use a function but make it local to what you're doing. Be careful to introduce new abstractions that other devs then have to learn. It is sometimes better to have something slightly more verbose that others will recognise than introducing some far away abstraction.

Thus perhaps:

```scala
def toMap(key: String, elems: Set[String]): Map[String, String] =
  if (elems.isEmpty) 
    Map.empty
  else
    Map(key -> elems.reduce(_ + "," + _))

val conductrFeatureArgs = toMap("CONDUCTR_FEATURES", featureNames)
val conductrRolesEnv = toMap("CONDUCTR_ROLES", conductrRoles)
```